src/panels/config/js/trigger/device.tsx

```diff
@@ -41,9 +88,76 @@ export default class DeviceTrigger extends Component<any, any> {
           hass={hass}
           label="Trigger"
         />
+        {showFor && (
+          <div>
+            <paper-input
+              id="device-trigger-days"
+              label="Days"
+              name="days"
+              type="number"
+              value={days}
+              onvalue-changed={this.daysChanged}
+            />
+            <paper-time-input
+              label={hass.localize(
+                "ui.panel.config.automation.editor.triggers.type.state.for"
+              )}
+              hour={hours}
+              min={minutes}
+              sec={seconds}
+              enable-second
+              format={24}
+              onhour-changed={this.hoursChanged}
+              onmin-changed={this.minutesChanged}
+              onsec-changed={this.secondsChanged}
+            />
+          </div>
+        )}
       </div>
     );
   }
+
+  private _pad(value) {
+    // Convert to zero-padded string for display
+    return parseInt(value, 10)
+      .toString()
+      .padStart(2, "0");
+  }
+
+  private _parseTime(trgFor) {
+    let days = "0";
+    let hours = "00";
+    let minutes = "00";
+    let seconds = "00";
+    if (trgFor && (trgFor.hours || trgFor.minutes || trgFor.seconds)) {
+      // If the trigger was defined using the yaml dict syntax, extract days, hours, minutes and seconds
+      ({ days = "0", hours = "00", minutes = "00", seconds = "00" } = trgFor);
+    }
+    if (typeof trgFor === "string" && trgFor.includes(":")) {
+      // Parse hours, minutes and seconds from ':'-delimited string
+      [hours = "00", minutes = "00", seconds = "00"] = trgFor.split(":");
```

Comment:
we would need to extract the hours above 23 to days

Reply:
I ditched support for time strings instead